PF/PF-Assignment-1/Shopping.java

```diff
@@ -0,0 +1,132 @@
+package sc;
+import java.util.ArrayList;
+import java.util.List;
+public class Shopping {
+       
+    private List<Item> Cart = new ArrayList<Item>();
+    
+    ShoppingCart(){
+    	
+    }
+ 
+    public void addItem(String Pname,float Pprice,int Pquantity )
+    {
+        Item Product= new Item(Pname,Pprice,Pquantity);
+        Cart.add(Product);
+    }
+    
+    public void removeItem(String Pname) {
+       Item delprod = null;
+       for (Item prod: Cart) {
+            if (prod.getPname() == Pname) {
+                Cart.remove(prod);
+                break;
+            }
+        }
+    }
+
+    public void updatePname(String OldName,String NewName)
+    {
+        for (Item prod: Cart) {
+            if (prod.getPname() == OldName) {
+                    prod.updatePname(NewName);
+            }
+        }
+    }
+    
+    public void updatePrice(String Pname, float NewPrice)
+    {
+        for (Item prod: Cart) {
+            if (prod.getPname() == Pname) {
+                    prod.updatePprice(NewPrice);
+            }
+        }
+    }
+    
+    public void updateQuantity(String Pname, int newQty)
+    {
+        for (Item prod: Cart) {
+            if (prod.getPname() == Pname) {
+                    prod.updatePqty(newQty);
+            }
+        }
+    }
+    
+    public void printCart()
+    {
+        for (Item prod: Cart) {
+            prod.printProduct();
+        }
+    }
+    
+    public void Bill()
```

Comment:
function name could be bit descriptive as 'Bill' seems vague. Can be named as 'generateBill' as it is more relevant to the context of the function